src/stack.cpp

```diff
@@ -294,7 +288,7 @@ static void slice_to_window_buf(HSSlice* s, struct s2wb* data) {
     }
 }
 
-void stack_to_window_buf(HSStack* stack, Window* buf, int len,
+void HSStack::to_window_buf(Window* buf, int len,
```

Comment:
I suggest fixing the signature of this method, replacing remain_len argument with a return value.

Reply:
Would you be okay with doing this separately?

Reply:
This will return a `vector<Window>` at some point, but I think the current version is fine for the time being.

Reply:
What kind of windows would this vector be comprised of?

Reply:
The windows in the stack. The monitor stack made things much more complicated than they need to be. Once the monitor stack is a simple vector (as a member of MonitorManager), the stack is a array (of length LAYER_COUNT) of HSSlice-vectors.